server.js

```diff
@@ -3107,6 +3107,100 @@ cache(function(data, match, sendBadge, request) {
   })}
 ));
 
+// DUB download integration
+camp.route(/^\/dub\/(dd|dw|dm|dt)\/([^\/]+)(?:\/([^\/]+))?\.(svg|png|gif|jpg|json)$/,
+cache(function (data, match, sendBadge, request) {
+  var info = match[1]; // downloads (dd - daily, dw - weekly, dm - monthly, dt - total)
+  var pkg = match[2]; // package name, e.g. vibe-d
+  var version = match[3]; // version (1.2.3 or latest)
+  var format = match[4];
+  var apiUrl = 'http://code.dlang.org/api/packages/'+pkg;
+  if (version)
+    apiUrl += '/' + version;
+  apiUrl += '/stats';
+  var badgeData = getBadgeData('dub', data);
+  request(apiUrl, function(err, res, buffer) {
+    if (err != null) {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(format, badgeData);
+    }
+    try {
+      var data = JSON.parse(buffer);
+      if (info.charAt(0) === 'd') {
```

Comment:
Hardly any reason, I wasn't sure that you can random index a string and charAt was the first thing the documentation showed.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String#Character_access
Looks like there used to be a reason and some people recommend charAt over the brackets because you can't mutate characters.
http://stackoverflow.com/questions/5943726/string-charatx-or-stringx

I can change it if you want me to.